src/safeposix/syscalls/net_calls.rs

```diff
@@ -1,5 +1,1505 @@
+                                } else {
+                                    drop(sockfdobj);
+                                    drop(filedesc_enum);
+                                    if self._nonblock_peek_read(*fd, fdtable) {
+                                        new_readfds.insert(*fd);
+                                        retval += 1;
+                                    }
+                                }
+                            }
+                        }
+
+                        //we don't support selecting streams
+                        Stream(_) => {continue;}
+
+                        //not supported yet
+                        Pipe(_) => {
+                            new_readfds.insert(*fd);
+                            retval += 1;
+                        }
+
+                        //these file reads never block
+                        _ => {
+                            new_readfds.insert(*fd);
+                            retval += 1;
+                        }
+                    }
+                } else {
+                    return syscall_error(Errno::EBADF, "select", "invalid file descriptor");
+                }
+            }
+
+            let fdtable = self.filedescriptortable.read().unwrap();
+            for fd in writefds.iter() {
+                if let Some(wrappedfd) = fdtable.get(&fd) {
+                    let mut filedesc_enum = wrappedfd.write().unwrap();
+                    match &mut *filedesc_enum {
+                        //we always say sockets are writable? Even though this is not true
+                        Socket(_) => {
+                            new_writefds.insert(*fd);
+                            retval += 1;
+                        }
+
+                        //we always say streams are writable?
+                        Stream(_) => {
+                            new_writefds.insert(*fd);
+                            retval += 1;
+                        }
+
+                        //not supported yet
+                        Pipe(_) => {
+                            new_readfds.insert(*fd);
+                            retval += 1;
+                        }
+
+                        //these file writes never block
+                        _ => {
+                            new_writefds.insert(*fd);
+                            retval += 1;
+                        }
+                    }
+                } else {
+                    return syscall_error(Errno::EBADF, "select", "invalid file descriptor");
+                }
+
+            }
+
+            //we'd do exceptfds here if we supported them
+
+            if retval != 0 || interface::readtimer(start_time) > end_time {
+                break;
+            } else {
+                interface::sleep(BLOCK_TIME);
+            }
+        }
+        *readfds = new_readfds;
+        *writefds = new_writefds;
+        return retval; //package out fd_set?
```

Comment:
Not sure what this comment is for or if it has to be addressed

Reply:
Ah this is referring to the fact that we deal with the fd_set stuff within the dispatcher rather than here, it probably should be removed?

Reply:
Or just have it clarify that.